File: Practice/DecreaseTheSumOfDigits.cpp
#include <vector>
#include <map>
#include <cstdlib>
#include <fstream>
#include <iomanip>   
#include <iostream>  
#include <sstream>  // istringstream buffer(myString);
#include <stack>
#include <algorithm>
#include <cstring>
#include <cassert>
#include <math.h>

// Useful constants 
#define INF                         (int)1e9 
#define EPS                         1e-9 

#define el '\n'

using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        long long n, s, sum, total;
        cin >> n >> s;
        string sN;
        sN = to_string(n);
        sum = 1;
        total = 0;
        int i = -1;
        while(sum <= s){
            i++;
            sum += sN.at(i) - '0';
        }
        total = pow(10, sN.length() - i) - n;
        cout << total << el;
    }
}

File: Practice/DecreaseTheSumOfDigits.py
t = int(input())
for k in range(t):
    n, s = map(int, input().split())
    sN = str(n)
    sum = 0
    total = 0
    i = 0
    while(i < len(sN)):
        sum += int(sN[i])
        i += 1
    if sum <= s:
        total = 0
    else:
        sum = 1
        i = -1
        while sum <= s:
            i += 1
            sum += int(sN[i])
        total = pow(10, len(sN) - i) - int(sN[i:])

    print(total)

